Apps/Sandcastle/gallery/Particle System Weather.html

```diff
@@ -0,0 +1,186 @@
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar"></div>
+<script id="cesium_sandcastle_script">
+    function startup(Cesium) {
+        'use strict';
+//Sandcastle_Begin
+/********* Set up the viewer, terrain, and camera *********/
+var viewer = new Cesium.Viewer('cesiumContainer', {
+    shouldAnimate : true
+});
+viewer.terrainProvider = new Cesium.CesiumTerrainProvider({
+    url : 'https://assets.agi.com/stk-terrain/v1/tilesets/world/tiles'
+});
+
+var scene = viewer.scene;
+scene.globe.depthTestAgainstTerrain = true;
+scene.camera.setView({
+    destination : new Cesium.Cartesian3(277096.634865404, 5647834.481964232, 2985563.7039122293),
+    orientation : {
+        heading : 4.731089976107251,
+        pitch : -0.32003481981370063
+    }
+});
+
+// temp variables for math
+var gravityScratch = new Cesium.Cartesian3();
+
+// snow
+var snowOptions = {
+    minParticleSize : scene.drawingBufferWidth / 60.0,
+    radius : 100000.0,
+
+    update : function(particle, dt) {
+        var position = particle.position;
+
+        Cesium.Cartesian3.normalize(position, gravityScratch);
+
+        var magnitude = -5500.0;
+        Cesium.Cartesian3.multiplyByScalar(gravityScratch, magnitude, gravityScratch);
+
+        particle.velocity = Cesium.Cartesian3.add(particle.velocity, gravityScratch, particle.velocity);
+    }
+};
+
+var snowProperties = {
+    modelMatrix : new Cesium.Matrix4.fromTranslation(scene.camera.position),
+    minimumSpeed : -2.0,
+    maximumSpeed : 2.0,
+    lifeTime : 5.0,
+    emitter : new Cesium.SphereEmitter(snowOptions.radius),
+    startScale : 0.5,
+    endScale : 1.0,
+    image : "https://cesium.com/blog/images/nts-snow-system/snow_particle.png",
+    rate : 5000.0,
+    startColor : Cesium.Color.WHITE.withAlpha(0.15),
+    endColor : Cesium.Color.WHITE.withAlpha(1),
+    minimumWidth : snowOptions.minParticleSize,
+    minimumHeight : snowOptions.minParticleSize,
+    maximumWidth : snowOptions.minParticleSize * 2.0,
+    maximumHeight : snowOptions.minParticleSize * 2.0,
+    forces : [snowOptions.update]
+};
+
+// rain
+var rainOptions = {
+    minParticleSize : scene.drawingBufferWidth / 80.0,
+    radius : 100000.0,
+
+    update : function(particle, dt) {
+        var position = particle.position;
+
+        Cesium.Cartesian3.normalize(position, gravityScratch);
+
+        var magnitude = -1050.0;
+        Cesium.Cartesian3.multiplyByScalar(gravityScratch, magnitude, gravityScratch);
+
+        particle.position = Cesium.Cartesian3.add(particle.position, gravityScratch, particle.position);
+    }
+};
+
+var rainProperties = {
+    modelMatrix : new Cesium.Matrix4.fromTranslation(scene.camera.position),
+    speed : 160.0,
+    lifeTime : 5.0,
+    emitter : new Cesium.SphereEmitter(rainOptions.radius),
+    startScale : 1.0,
+    endScale : 0.0,
+    image : "https://cesium.com/blog/images/nts-snow-system/snow_particle.png",
```

Comment:
Should the rain particles use a different image maybe? 

Also can we make them a little more translucent, and a slightly darker blue?

Reply:
Another cool thing could maybe be to fade the color depending on distance of the particle from the camera.